src/user_main.c

```diff
@@ -98,6 +102,67 @@ void ICACHE_FLASH_ATTR  gpio16_output_set(uint8 value)
         (READ_PERI_REG(RTC_GPIO_OUT) & (uint32)0xfffffffe) | (uint32)(value & 1)
     );
 }
+void beep(int ms)
+{
+    gpio16_output_set(1);
+    os_delay_us(ms*1000);
+    gpio16_output_set(0);
+}
+
+void enableArc()
+{
+    if (GPIO_INPUT_GET(4) == 0) {
+        GPIO_OUTPUT_SET(4,1);
+        beep(500);
+    }
+}
+
+void disableArc()
+{
+    if (GPIO_INPUT_GET(4) == 1) {
+        GPIO_OUTPUT_SET(4,0);
+    }
+}
+
+void updatePwm()
+{
+    if (adc_pedal == 0) {
+        disableArc();
+        pwm_duty = adc_pot / (1000.0/PWM_PERIOD);
+    } else {
+        pwm_duty = (adc_pot * adc_pedal) / (1000000.0/PWM_PERIOD);
```

Comment:
Am I missing something here? By calculations max pot and pedal states would give just around 11-12% of pwm range:
(1000 * 1024) / (100000 / 5000) = 512 which would be 512 ticks ON of PWM_PERIOD (5000)

Reply:
Nvermind 1 000 000 / 5000 not 100 000.... This overflows pwm a bit, but library should cap the value

Reply:
Or you could increase PWM_PERIOD to 5120 that would give pwm frequency of ~976.5 Hz But full range of your pwm math

Reply:
I normalize both ADC values here to a max of 1000 trimming any overflow so PWM should max out at 5000

Reply:
Aaah yes. Missed normalization of pot adc...